keps/sig-node/4815-dra-partitionable-devices/README.md

```diff
@@ -0,0 +1,2093 @@
+to include a set of mixins in a device definition, and (2) the ability to
+express that capacity from one device gets consumed by another device if/when
+the scheduler decides to allocate it.
+
+To simplify the conversation, we discuss each new feature separately, starting
+with "mixins" and the new `Includes` field, which allows a set of mixins to
+extend a device with common attributes and capacities.
+
+### Extending a device with as set of mixins
+
+A simple example the defines a set of mixins and includes them in the
+definition of 4 NVIDIA A100 GPUs can be seen below:
+
+```yaml
+deviceMixins:
+- name: system-attributes
+  composite:
+    attributes:
+      cudaDriverVersion:
+        version: 12.6.0
+      driverVersion:
+        version: 560.35.3
+- name: common-gpu-attributes
+  composite:
+    attributes:
+      type:
+        string: gpu
+      architecture:
+        string: Ampere
+      brand:
+        string: Nvidia
+      productName:
+        string: NVIDIA A100-SXM4-40GB
+      cudaComputeCapability:
+        version: 8.0.0
+- name: common-gpu-capacities
+  composite:
+    capacity:
+      memory:
+        quantity: 40Gi
+devices:
+- name: gpu-0
+  composite:
+    includes:
+    - name: system-attributes
+    - name: common-gpu-attributes
+    - name: common-gpu-capacities
+    attributes:
+      index:
+        int: 0
+      minor:
+        int: 0
+      uuid:
+        string: GPU-4cf8db2d-06c0-7d70-1a51-e59b25b2c16c
+- name: gpu-1
+  composite:
+    includes:
+    - name: system-attributes
+    - name: common-gpu-attributes
+    - name: common-gpu-capacities
+    attributes:
+      index:
+        int: 1
+      minor:
+        int: 1
+      uuid:
+        string: GPU-4404041a-04cf-1ccf-9e70-f139a9b1e23c
+- name: gpu-2
+  composite:
+    includes:
+    - name: system-attributes
+    - name: common-gpu-attributes
+    - name: common-gpu-capacities
+    attributes:
+      index:
+        int: 2
+      minor:
+        int: 2
+      uuid:
+        string: GPU-79a2ba02-a537-ccbf-2965-8e9d90c0bd54
+- name: gpu-3
+  composite:
+    includes:
+    - name: system-attributes
+    - name: common-gpu-attributes
+    - name: common-gpu-capacities
+    attributes:
+      index:
+        int: 3
+      minor:
+        int: 3
+      uuid:
+        string: GPU-662077db-fa3f-0d8f-9502-21ab0ef058a2
+```
+
+As you can see, three mixins are created called "system-attributes",
+"common-gpu-attributes", and "common-gpu-capacities", which all get included in
+the definitons of the actual GPU devices themselves, along with their own
+device-specific attributes (and device-specific capacities if there were any).
+
+With this in place, the scheduler can parse these device definitions and
+"flatten" them into something that looks exactly the same as a `Basic` device.
+The scheduler can then allocate such devices using the same algorithm it uses
+for `Basic` devices.
+
+The code to do this flattening is fairly straightforward, and can be
```

Comment:
If it's so straightforward, it probably shouldn't be included in the KEP, for brevity.

Reply:
OK

Reply:
Removed